app/controllers/application_controller.rb

```diff
@@ -131,6 +136,7 @@ def skip_session_expiration
 
   def set_locale
     I18n.locale =
+      params[:locale] ||
```

Comment:
This still needs to be moved to a method to check against available locales, to avoid the `InvalidLocale` exception.